### ISSUE TITLE
Sprint plan for Fabric8 Analytics: #144

### ISSUE THREAD

**Issue by krishnapaparaju** [openshiftio/openshift.io#1971]
**IDE Extensions: Make required changes as per PM's feedback**

**Extend insights to cover WildFly Swarm applications**

- [ ] #2090 

**TODOs for publishing Devstudio plugin**

- [x] Replace all printStackTrace and System.out.println calls with proper logging(https://github.com/fabric8-analytics/fabric8-analytics-devstudio-plugin/issues/149)
- [x] Refactor plugins(https://github.com/fabric8-analytics/fabric8-analytics-devstudio-plugin/issues/143)
- [x]   Send license file to Stack Analyses(https://github.com/fabric8-analytics/fabric8-analytics-devstudio-plugin/issues/156)

**Continue working towards Combined wizard: Single component for Wizard and Launcher operational at OSIO, launch.openshift.io**

- [x] Need to set up daily standup with all the teams involved, as currently end to end work plan, expected timelines are not in place.
- [x] Getting Started integration related activities - https://github.com/openshiftio/openshift.io/issues/2178

**Continue working towards Analytics driven Dependency Editor be integrated into OSIO Wizard**

- [x] Implement the designed UX mockup with an ability to test as standalone application (till its integrated into wizard)

- [x] Implement necessary changes to the API side: External routes for each functionality supported as part of the one big API (API for Stack report)

- [x] Target end to end flow (standalone app.) for the sprint demo (144)

**Launch Che workspace with the generated project from combined wizard**

- [ ] Work with Che team, Stevan to understand the Che file support at OSIO

**POC, UX for application stack editor, supporting multi container applications**

- [ ] Work with Stevan to refine the specification for application stack editor

- [ ] Start with POC for application stack editor

**Integrate insights into Che editor**
- [x] Current POC:  Get feedback from PM team

- [ ] Work with UX team to finalize the UX changes

- [ ] Integrate the changes to LSP and deploy changes at OSIO

**Continue working towards Scaling the platform to support 1000 concurrent users** (#1144)

- [x] Document the test results

- [ ] Work with SD team to get necessary resources

- [ ] Promote changes to OSIO

**Continue working towards  OSIO Gemini related activities**

- [x] Define the end to end flow around various components involved at Fabric8 Analytics platform

- [ ] Identify the key differences (end to end flow) for an Gemini users (a OSIO developer team) and OSIO users 

- [ ] Identify integration points, UX for OSIO user to visualize the currently used application dependencies, get updates around his/her (OSIO user) application stack

- [x] Work with Chmouel, SD teams  to work thru setting up build pipelines

- [ ] Target Build pipelines, registry infra. be in place 

- [x] Work with Bama to start integrating Fabric8 Analytics with a new Scanner

**Continue working towards Map CVEs for the Maven, Node ecosystems**

- [x] Continue the current manual mapping from NVDB to individual packages

- [x] Continue working towards POC for automating CPEs to individual dependencies #2106 

**Service design, deployment for ingesting manifest files**
   

- [ ] Identify, Implement API end points (not user facing) for triggering ingesting manifest files with an associated criteria (eg: Python manifest files at GitHub with > 50 GitHub stars)
 

**Improve service side for better handle on service deployment** #1087 

- [ ] #1084 

- [ ] #1083 

- [x] Extend #1085 to cover complete Maven and Node ecosystems

- [x] #1330 

<img width="27" alt="screen shot 2018-01-19 at 6 04 16 pm" src="https://user-images.githubusercontent.com/15967765/35151147-3731edb6-fd43-11e7-9fdb-da70b2f55db0.png"> **[Stretch goal]  Continue working towards Identifying important functions at a code base, generate AI based unit tests for these functions** #1792 

- [ ] Get to a language neutral form: Algorithm to convert ASTs (for Java) to DSLs

<img width="27" alt="screen shot 2018-01-19 at 6 04 16 pm" src="https://user-images.githubusercontent.com/15967765/35151147-3731edb6-fd43-11e7-9fdb-da70b2f55db0.png"> **[Stretch goal] Continue working towards supporting Insights for Node ecosystem** #1809  

- [x] Implement 'CVAE' and document insights been generated along with accuracy (manual testing to start with) (#2002 and #2004 )

- [ ] Implement 'Hierarchical Poisson Factorization' and document insights been generated along with accuracy (manual testing to start with)

**Comment by nickboldt** [comment]
Question.... should there be checkboxes under "TODOs for publishing Devstudio plugin" ? 

When do you plan to have something stable, with passing tests, ready to add to jbosstools/devstudio Central? Next code freeze (for 4.5.3.AM1 / 11.3.0.AM1) is Feb 7, the last Wednesday of sprint 144. See Google [1] and JIRA calendars [2], [3] for more details and to know when subsequent milestones/code freezes are released. (Spoiler alert, it's every 3 weeks!)

So it should be on or before that day, if you want something released to the public for this sprint. Note that it'll only be public for JBoss Tools users as we don't publish pre-GA devstudio milestones. 

[1] https://bitly.com/rhdtrain
[2] https://issues.jboss.org/projects/JBDS?selectedItem=com.atlassian.jira.jira-projects-plugin:release-page
[3] https://issues.jboss.org/projects/JBIDE?selectedItem=com.atlassian.jira.jira-projects-plugin:release-page

**Comment by krishnapaparaju** [comment]
@nickboldt +1

@sivaavkd can you please refine work items related to DevStudio plugin work..

**Comment by nickboldt** [comment]
[ ] Enable tests. Verify all tests pass in Jenkins
[ ] Ensure back end service requirements are stable
[ ] QE approves contribution for inclusion in JBoss Tools and Devstudio Central, as a Technology Preview. 
[ ] Declaration of readiness is posted to the JIRA: https://issues.jboss.org/browse/JBIDE-24897 
[ ] Nick publishes a copy of the latest CI build of the fabric8 analytics plugins & feature to its own "milestone" update site. NOTE: In future this step can be done by Geetika or someone else on the plugin development team, to update the site a newer milestone version, and to ultimately release a .Final release once ready. This is accomplished via a Jenkins job, so it's really simple.
[ ] Nick adds the feature to Central, using the new milestone site, and publishes an updated Discovery Site
[ ] QE tests the Central Discovery Site against the latest jbosstools/devstudio CI build
[ ] If approved, the feature stays in Central and is available as of the next jbosstools milestone release (every 3 weeks) or the next devstudio GA release (every 12 weeks)
[ ] Once the plugin is ready to be considered STABLE (rather than a Technology Preview milestone) Nick will update the URL used in Central to a /stable/ instead of /development/ one. This should be timed to occur before the latest upcoming code freeze.

**Comment by GeetikaBatra** [comment]
@nickboldt and @krishnapaparaju 
Following are the items left from development point of view.

- [ ] Replace all printStackTrace and System.out.println calls with proper logging(fabric8-analytics/fabric8-analytics-devstudio-plugin#149)
- [ ] Refactor plugins(fabric8-analytics/fabric8-analytics-devstudio-plugin#143)

**Comment by nickboldt** [comment]
I'm ready to add F8A to Central. https://github.com/jbosstools/jbosstools-discovery/pull/375

But first we need to publish a milestone release to its own update site: http://download.jboss.org/jbosstools/oxygen/development/updates/fabric8analytics/

Let me know which build you'd like tagged as a milestone, from those here:

http://download.jboss.org/jbosstools/oxygen/snapshots/builds/jbosstools-fabric8analytics_master/

**Comment by rawagner** [comment]
@nickboldt B701 please http://download.jboss.org/jbosstools/oxygen/snapshots/builds/jbosstools-fabric8analytics_master/2018-01-31_11-56-20-B701/

**Comment by nickboldt** [comment]
Done. http://download.jboss.org/jbosstools/oxygen/development/updates/core/fabric8analytics/

**Comment by nickboldt** [comment]
Please note that " Ensure back end service requirements are stable" also includes 'does the website work with Internet Explorer / default OOTB embedded Eclipse browser?' which apparently... is no, not yet. :)

**Comment by GeetikaBatra** [comment]
@nickboldt, it is working on Windows IE. 
https://issues.jboss.org/browse/JBIDE-24897?focusedCommentId=13527755&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13527755

Things are complete from our end. I guess we are good to go.

**Comment by nickboldt** [comment]
Sounds good. 

Next steps:

- [ ] QE tests the Central Discovery Site against the latest jbosstools/devstudio CI build
- [ ] If approved, the feature stays in Central and is available as of the next jbosstools milestone release (every 3 weeks) or the next devstudio GA release (every 12 weeks)
- [ ] Once the plugin is ready to be considered STABLE (rather than a Technology Preview milestone) Nick will update the URL used in Central to a /stable/ instead of /development/ one. This should be timed to occur before the latest upcoming code freeze.

**Comment by sivaavkd** [comment]
Closing 144 planning.  We have moved any open items to 145.
